common/diagnostics/Metrics.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.core.common.diagnostics;
+
+import com.eclipsesource.json.JsonObject;
+
+import java.util.TimeZone;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Metrics {
+    private final SystemProperties system;
+    private final NetworkRequests requests;
+    private final UsageCounters usage;
+    private final UserErrorStatistics userErrors;
+
+    public Metrics(String serverID, String name, String version) {
+        this.system = new SystemProperties(serverID, name, version);
+        this.requests = new NetworkRequests();
+        this.usage = new UsageCounters();
+        this.userErrors = new UserErrorStatistics();
+    }
+
+    public void requestAttempt(Metrics.NetworkRequests.Kind kind) {
+        requests.attempt(kind);
+    }
+
+    public void requestSuccess(Metrics.NetworkRequests.Kind kind) {
+        requests.success(kind);
+    }
+
+    public void setCurrentCount(UsageCounters.Kind kind, long value) {
```

Comment:
the other methods are called `<type>Attempt/success/etc.` and this one is mismatched between method name and counter type, is that intended? maybe `setUsageCount`?

Reply:
`setCurrentCount(DATABASES, count)` reads better on callsite. Renamed `UsageCounters` to `CurrentCounts`.